cmd/api/src/services/fileupload/file_upload.go

```diff
@@ -82,13 +85,22 @@ func GetFileUploadJobByID(db FileUploadData, jobID int64) (model.FileUploadJob,
 	return db.GetFileUploadJob(jobID)
 }
 
-func SaveToTempFile(location string, fileData io.ReadCloser) (string, error) {
-	if tempFile, err := os.CreateTemp(location, "bh"); err != nil {
-		return "", fmt.Errorf("error creating temp file: %w", err)
-	} else if _, err := io.Copy(tempFile, fileData); err != nil {
-		return "", fmt.Errorf("error copying temp file: %w", err)
+func SaveIngestFile(location string, fileData io.ReadCloser) (string, error) {
+	var typeBuf = make([]byte, 512)
+	tempFile, err := os.CreateTemp(location, "bh")
+	if err != nil {
+		return "", fmt.Errorf("error creating ingest file: %w", err)
+	}
+	defer tempFile.Close()
+	if _, err := io.Copy(tempFile, fileData); err != nil {
```

Comment:
Not sure if it's worth the effort depending on how often we expect invalid files, but you could use `io.CopyN()` or `io.ReadFull()` or something to get the first 512 bytes of the file and do the type detection immediately instead of having to pipe the entire request out to disk before you can go back and check if it's valid or not. Then if it's valid you can just write the buffer out to the file and then use `io.Copy()` to copy the rest like before.

This would also allow us to not even create the temporary file until after the content type checking, thus avoiding some extra files on disk (similar to the last comment about deleting temporary files)

Reply:
I think it's fine for now. We currently accept all files already (we have an orphan cleanup process), and since this is more of a client feedback improvement and less of an ingest validation improvement (which we we have a ticket open), I believe we can leave as is.